base/utils/R/download.url.R

```diff
@@ -23,16 +23,16 @@
 download.url <- function(url, file, timeout = 600, .opts = list(), retry404 = TRUE) {
   dir.create(basename(file), recursive = TRUE)
   count <- 0
-  while (!RCurl::url.exists(url, .opts = .opts) && count < timeout) {
+  while (!curl::curl_fetch_memory(url, handle = curl::new_handle(nobody = 1)) && count < timeout) {
```

Comment:
I see two subtleties here:
* When the url doesn't exist, `curl_fetch memory` will still return a value that evaluates to TRUE, so we need to explicitly check the status code (e.g. `(curl_fetch_memory(...)$status_code %/% 200) == 1`)
* More trickily, when the requested hostname can't be resolved `curl_fetch_memory` will throw an error whereas `RCurl::url.exists` just returns FALSE. I'm not sure whether we need to handle this case or not but will ask around.

Reply:
@infotroph should we be using httr:HEAD then?

Reply:
Looks like httr::HEAD has the same behavior both with respect to status codes and with respect to erroring on unresolvable hostnames, so we don't need to switch because of those. However HEAD definitely shows the intention much clearly than the `nobody = 1` incantation does, so it'd be worth using at least in packages that already import httr.